extensions/panache/hibernate-reactive-panache-common/runtime/src/main/java/io/quarkus/hibernate/reactive/panache/common/runtime/CommonPanacheQueryImpl.java

```diff
@@ -69,6 +69,35 @@ public <T> CommonPanacheQueryImpl<T> project(Class<T> type) {
             throw new PanacheQueryException("Unable to perform a projection on a named query");
         }
 
+        String lowerCasedTrimmedQuery = query.trim().toLowerCase();
+        if (lowerCasedTrimmedQuery.startsWith("select new ")) {
+            throw new PanacheQueryException("Unable to perform a projection on a 'select new' query: " + query);
+        }
+
+        // If the query starts with a select clause, we generate an HQL query
+        // using the fields in the select clause:
+        // Initial query: select e.field1, e.field2 from EntityClass e
+        // New query: SELECT new org.acme.ProjectionClass(e.field1, e.field2) from EntityClass e
+        if (lowerCasedTrimmedQuery.startsWith("select ")) {
```

Comment:
Can't say I'm a big fan of all these string manipulations. My guess is that it's going to be brittle and we will get bitten by this at some point but I suppose we will see.

Reply:
I'm not going to write a compiler if that's what you are suggesting

Reply:
I'm not really suggesting that for now. As I said, let's see how it goes.